pointcloud/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSource.scala

```diff
@@ -0,0 +1,126 @@
+ * limitations under the License.
+ */
+
+package geotrellis.pointcloud.raster.ept
+
+import geotrellis.pointcloud.raster.rasterize.triangles.PDALTrianglesRasterizer
+import geotrellis.proj4._
+import geotrellis.raster._
+import geotrellis.raster.interpolation._
+import geotrellis.raster.io.geotiff.OverviewStrategy
+import geotrellis.vector._
+
+import cats.syntax.option._
+import _root_.io.circe.syntax._
+import _root_.io.pdal.pipeline._
+import org.log4s._
+import spire.syntax.cfor._
+
+import scala.collection.JavaConverters._
+
+/**
+  * [[DEMRasterSource]] doesn't use [[OverviewStrategy]].
+  * At this point, it relies on the EPTReader logic:
+  * https://github.com/PDAL/PDAL/blob/2.1.0/io/EptReader.cpp#L293-L318
+  */
+case class IDWRasterSource(
+  path: EPTPath,
+  resampleTarget: ResampleTarget = DefaultTarget,
+  sourceMetadata: Option[EPTMetadata] = None,
+  threads: Option[Int] = None,
+  targetCellType: Option[TargetCellType] = None
+) extends RasterSource {
+  @transient private[this] lazy val logger = getLogger
+
+  lazy val metadata: EPTMetadata = sourceMetadata.getOrElse(EPTMetadata(path.value))
+
+  def attributes: Map[String, String] = metadata.attributes
+  def attributesForBand(band: Int): Map[String, String] = metadata.attributesForBand(band)
+  def bandCount: Int = metadata.bandCount
+  def cellType: CellType = metadata.cellType
+  def crs: CRS = metadata.crs
+  def gridExtent: GridExtent[Long] = metadata.gridExtent
+  def name: SourceName = metadata.name
+  def resolutions: List[CellSize] = metadata.resolutions
+
+  def reprojection(targetCRS: CRS, resampleTarget: ResampleTarget, method: ResampleMethod, strategy: OverviewStrategy): IDWReprojectRasterSource =
+    IDWReprojectRasterSource(path.value, targetCRS, resampleTarget, sourceMetadata = metadata.some, threads = threads, method, targetCellType = targetCellType)
+
+  def resample(resampleTarget: ResampleTarget, method: ResampleMethod, strategy: OverviewStrategy): IDWResampleRasterSource =
+    IDWResampleRasterSource(path.value, resampleTarget, metadata.some, threads, method, targetCellType)
+
+  def read(bounds: GridBounds[Long], bands: Seq[Int]): Option[Raster[MultibandTile]] = {
+    val targetRegion = gridExtent.extentFor(bounds, clamp = false)
+    val Extent(exmin, eymin, exmax, eymax) = targetRegion.extent
+
+    val expression = ReadEpt(
+      filename   = path.value,
+      resolution = gridExtent.cellSize.resolution.some,
+      bounds     = s"([$exmin, $eymin], [$exmax, $eymax])".some,
+      threads    = threads
+    )
+
+    logger.debug(expression.asJson.spaces4)
+
+    val pipeline = expression toPipeline
+
+    try {
+      if(pipeline.validate()) {
+        pipeline.execute
+
+        val pointViews = pipeline.getPointViews().asScala.toList
+        assert(pointViews.length == 1, "Triangulation pipeline should have single resulting point view")
+
+        pointViews.headOption.map { pv =>
+          val pc = pv.getPointCloud
```

Comment:
It looks to me this could be moved into a seaprate function / package / object. Otherwise it is just copypasted across Resample / Reproject / Common one raster sources.